### PR TITLE
Remove msiexec dependency by adding using `vcpkg.exe` to unpack `7z.exe`

### PR DESCRIPTION
**Pull request by JavierMatosD**
Alternative to https://github.com/microsoft/vcpkg/pull/41042

### CODE REVIEW

**Comment by JavierMatosD** [comment]
DO NOT MERGE: This PR depends on a tool release

File: scripts/cmake/vcpkg_extract_archive.cmake [hunk 1]
```diff
@@ -34,12 +34,8 @@ function(vcpkg_extract_archive)
             WORKING_DIRECTORY "${archive_directory}"
         )
     elseif("${archive_extension}" MATCHES [[\.7z\.exe$]])
-        vcpkg_find_acquire_program(7Z)
         vcpkg_execute_in_download_mode(
-            COMMAND ${7Z} x
-                "${arg_ARCHIVE}"
-                "-o${arg_DESTINATION}"
-                -y -bso0 -bsp0
+            COMMAND "$ENV{VCPKG_COMMAND}" z-extract "${arg_ARCHIVE}" "${arg_DESTINATION}"
```
Comment (Neumann-A) [inline scripts/cmake/vcpkg_extract_archive.cmake:38]: $ENV ?

Comment (JavierMatosD) [reply scripts/cmake/vcpkg_extract_archive.cmake:38]: It's how it is invoked in `vcpkg_download_distfile`? Is there some concern?

Comment (Neumann-A) [reply scripts/cmake/vcpkg_extract_archive.cmake:38]: I am just suprised to see it as an env variable. Would have expected it to be passed as a normal cmake variable.

File: scripts/azure-pipelines/azure-pipelines.yml [hunk 2]
```diff
@@ -8,7 +8,7 @@ parameters:
   - name: vcpkgToolSha
     displayName: 'Custom SHA of vcpkg-tool to use rather than bootstrap'
     type: string
-    default: 'use default'
+    default: 7869ef3adbc5225037951bb06f2edb9506f9a7a5
```
Comment (data-queue) [inline scripts/azure-pipelines/azure-pipelines.yml:11]: note: change this back before merging

**Review by data-queue — DISMISSED** [review]

**Pull request merged**